### PR TITLE
Improve error message when user script errors during runtime

### PR DESCRIPTION
**Pull request by defunctzombie**
**User-Facing Changes**
When a user script errors, the new message displayed to the user is the following:

Screen Shot 2022-08-17 at 6 09 36 PM

Additionally, the `Problems` tab diagnostics list is fixed to removed the `undefined` text that would show up after `Runtime`.

Screen Shot 2022-08-17 at 5 45 12 PM

**Description**
When a user script errors during runtime and fails to produce a message, the previous error displayed would only indicate that a script failed to produce a message. This warning did not make it obvious that the user script in fact errored during runtime.

This change improves the error to be clear that the script errored during runtime when there is an error rather than only a missing message.

Fixes: #4084

### CODE REVIEW

**Comment by amacneil** [comment]
Screenshot looks ok for now. One day we can figure out a better place to expose playback errors like this.

**Comment by defunctzombie** [comment]
> Screenshot looks ok for now. One day we can figure out a better place to expose playback errors like this.

We could add dots on the timeline if we want - just seems generally more invasive /shrug

File: packages/studio-base/src/players/UserNodePlayer/index.ts [hunk 1]
```diff
@@ -485,26 +485,34 @@ export default class UserNodePlayer implements Player {
...
+            code: ErrorCodes.RUNTIME,
+          });
         }
+
         this._addUserNodeLogs(nodeId, result.userNodeLogs);
 
+        if (allDiagnostics.length > 0) {
+          this._problemStore.set(problemKey, {
+            severity: "error",
+            message: `User Script ${nodeId} suffered an error.`,
```
Comment (jtbandes) [inline packages/studio-base/src/players/UserNodePlayer/index.ts:503]:
```suggestion
            message: `User Script ${nodeId} encountered an error.`,
```

**Review by jtbandes — APPROVED** [review]

File: packages/studio-base/src/players/UserNodePlayer/index.ts (hunk 1 above)
Comment (defunctzombie) [reply packages/studio-base/src/players/UserNodePlayer/index.ts]: I accepted this change but I do feel like suffered is better :D encountered is like the script ran into the error but the error is something that was along the scripts journey. Suffered is more like the script itself became ill with the error 🤣

**Comment by defunctzombie** [comment]
> I think we should get rid of "No details provided". It doesn't help add any context for the user.

Sure - I can make that change in the dialog. If there is a `tip` but no `error` I can remove the no details provided. I think it is a useful fallback when there is no tip _and_ no error.

**Pull request merged**